bfdd/bfd.c

```diff
@@ -1800,6 +1807,67 @@ void bfd_profiles_remove(void)
 		bfd_profile_free(bp);
 }
 
+struct _bfd_session_has_echo {
+	/* VRF peers must match */
+	struct vrf *vrf;
+	/* Echo enabled or not */
+	bool enabled;
+};
+
+static int _bfd_session_has_echo(struct hash_bucket *hb, void *arg)
+{
+	const struct bfd_session *session = hb->data;
+	struct _bfd_session_has_echo *has_echo = arg;
+
+	if (session->vrf != has_echo->vrf)
+		return HASHWALK_CONTINUE;
+	if (!CHECK_FLAG(session->flags, BFD_SESS_FLAG_ECHO))
+		return HASHWALK_CONTINUE;
+
+	has_echo->enabled = true;
+	return HASHWALK_ABORT;
+}
+
+void bfd_vrf_toggle_echo(struct bfd_vrf_global *bfd_vrf)
+{
+	struct _bfd_session_has_echo has_echo = {};
```

Comment:
shouldn't we set the correct vrf in has_echo?

Reply:
Yes, thank you. I missed that bit.